test/tools/yulInterpreter/EWasmBuiltinInterpreter.cpp

```diff
@@ -133,6 +135,12 @@ u256 EWasmBuiltinInterpreter::evalBuiltin(YulString _fun, vector<u256> const& _a
 		writeMemoryWord(arg[0], arg[1]);
 		return 0;
 	}
+	else if (_fun == "i64.store8"_yulstring)
+	{
+		accessMemory(arg[0], 1);
+		writeMemoryByte(arg[0], uint8_t(arg[1] >> 56));
```

Comment:
Why is this taking the top 8 bits? Are the stack items not native-endian in the interpreter?

Reply:
Implementation of mstore8 in polyfill in PR https://github.com/ethereum/solidity/pull/7766 makes endian_swap on i64 value which is not needed. Assuming that it should be lowest 8 bits. Am I right?

Reply:
Wasm is little endian when writing to memory, but the input is native endian (in the interpreter anyway), so it should take the lowest 8 bits. Also since we're writing 8 bits in total there should be no endianness question here.

Reply:
Can you also use `static_cast<uint8_t>(arg[1] & 0xff)` here?